spyder/widgets/explorer.py

```diff
@@ -541,11 +545,14 @@ def delete(self, fnames=None):
         multiple = len(fnames) > 1
         yes_to_all = None
         for fname in fnames:
-            yes_to_all = self.delete_file(fname, multiple, yes_to_all)
-            if yes_to_all is not None and not yes_to_all:
-                # Canceled
-                return
-
+            if osp.isdir(fname) and osp.exists(osp.join(fname,'.spyproject')):
+                self.delete_project(fname)
+            else:    
+                yes_to_all = self.delete_file(fname, multiple, yes_to_all)
+                if yes_to_all is not None and not yes_to_all:
+                    # Canceled
+                    break
+                

```

Comment:
Sorry, since `delete_project` in `plugins/projects.py` doesn't receive any arguments, the logic is

```python
if fname == self.root_path:
    self.delete_project.emit()
elif '.spyproject' in fname:
    self.delete_project.emit()
else:
   ...
```

```python
delete_project = Signal()
```